mp/src/game/server/ai_memory.cpp

```diff
@@ -546,7 +546,11 @@ bool CAI_Enemies::HasFreeKnowledgeOf( CBaseEntity *pEnemy )
 
 	if ( pEnemy != AI_UNKNOWN_ENEMY )
 		DevWarning( 2,"Asking HasFreeKnowledgeOf for enemy that's not in my memory!!\n");
+#ifdef NEO
+	return false;
```

Comment:
I think we should return `true` here, because the base engine behaviour is casting `AI_INVALID_TIME` to bool, and it evaluates to a nonzero value. And any nonzero boolean has the truth value of `true`, not `false`. Unless there's some specific reason for us to differ from the base engine implementation here?

Reply:
My rational was it seems like the usage of `AI_INVALID_TIME` was an accidental mistake and was intend to be false. Although you are right it was basically a true originally.

Reply:
I've opted to do the static_cast route and put a comment on it regarding this line.

Reply:
Cool, yeah I think semantically this PR should probably be a pure refactor. If this particular behaviour ever becomes an issue, we should probably handle it separately.